DataSet.rst

```diff
@@ -512,76 +573,154 @@ year, month, day, hour, minute, second, millisecond.
+An API is provided in pbcore_ (Python) and pbbam_ (C++) that makes consuming DataSet XML files or the
+underlying files such as BAM as burden-free as possible.
 
 
-Other Bioinformatics Tools
-++++++++++++++++++++++++++
+Other PacBio command-line tools
+--------------------------------
 
 All bioinformatics tools that consume DataSet XML files should be
 capable of producing identical results using the equivalent BAM or
 FASTA file generated after applying all filters. In other words, the
 DataSet XML is not required to obtain bioinformatics results. However,
-`support` for the DataSet XML is encouraged for Secondary Analysis tools,
+`support` for the DataSet XML is encouraged for SMRT Analysis tools,
 and facilitated using the above API.
 
-SMRT Portal and SMRT Pipe
-+++++++++++++++++++++++++
+
+SMRT Link and SMRT Pipe
+-----------------------
 
 The DataSet XML is required for display of DataSets (such as
-references) in SMRT Portal and for chunking in the distributed pipelines
+references) in SMRT Link and for chunking in the distributed pipelines
 using pbsmrtpipe. Moreover, for these applications the DataSetMetadata
 field is mandatory, not optional.
 
-3.7 DataSet mutability and equality
------------------------------------
+
+Other topics
+==============
+
+
+Mutability and equality
+-----------------------
 To allow user editing of attributes such as Name without affecting the
 underlying DataSet we define the Core DataSet as the XML with the user
 editable attributes (Name, Description and Tags) removed (not set to "",
 but absent).  This Core DataSet is immutable and is the entity on which
-identity operations will be defined. As a consequence, any modifications
+identity operations are defined. As a consequence, any modifications
 to fields other than Name, Description or Tags requires giving the DataSet
 a new UniqueId. Operations such as md5 checksum should be performed on
 the Core DataSet unless otherwise specified.
 
 
-4. Outstanding Issues and Future Directions
-===========================================
+I/O trade-offs
+--------------
+The DataSet model defers I/O operations by replacing up-front file merges
+with downstream I/O operations that hit many different files. This allows
+consumers to avoid explicit creation of files on disk and the resulting
+redundant storage and costly write operations. For many uses this many-file
+approach is preferred to explicitly creating the file on disk,
+but in some cases it may be desirable to incur the cost of accessing
+and filtering multiple files once (e.g. to reduce disk seeks for highly
+fragmented DataSets). Determining when the costs outweigh the benefits will
+need practical investigation, but regardless the Consolidate operation
+provides the means for using the form of DataSet that best fits a
+particular use case.
+
+
+Outstanding Issues and Future Directions
+---------------------------------------------
+
+- Document FASTA filters for pbcore / pbbam for releases post-4.0.
```

Comment:
It sounds a bit like scope creep to me to have such a thing... but I could be wrong